workers/generic-worker/d2g_test.go

```diff
@@ -508,3 +510,182 @@ func TestD2GHostSharedMemory(t *testing.T) {
+		Region            string `json:"region"`
+	}
+	PartialTask struct {
+		ProvisionerID string `json:"provisionerId"`
+		Payload       PartialPayload
+	}
+	PartialPayload struct {
+		// This ensures to test that the Docker Worker command is included,
+		// rather than the Generic Worker native command generated by D2G.
+		Command []string `json:"command"`
+	}
+)
+
+func TestD2GChainOfTrustNamedDockerImage(t *testing.T) {
+
+	setup(t)
+	image := d2g.NamedDockerImage{
+		Name: "taskcluster/taskcluster-proxy:v81.0.2",
+		Type: "docker-image",
+	}
+
+	expected := DockerWorkerPartialChainOfTrustCert{
+		Environment: PartialEnvironment{
+			ImageHash:         "sha256:d1c5c701d837b59386563dc702932d200adc41f8bd99a1de451e6fcf378dbfa5",
+			ImageArtifactHash: "",
+			Region:            "test-worker-group",
+		},
+		Task: PartialTask{
+			ProvisionerID: "test-provisioner",
+			Payload: PartialPayload{
+				Command: []string{
+					"taskcluster-proxy",
+					"--version",
+				},
+			},
+		},
+	}
+
+	D2GChainOfTrustHelper(t, &image, []string{}, expected)
+}
+
+func TestD2GChainOfTrustDockerImageName(t *testing.T) {
+
+	setup(t)
+	image := d2g.DockerImageName("taskcluster/taskcluster-proxy:v81.0.2")
+
+	expected := DockerWorkerPartialChainOfTrustCert{
+		Environment: PartialEnvironment{
+			ImageHash:         "sha256:d1c5c701d837b59386563dc702932d200adc41f8bd99a1de451e6fcf378dbfa5",
+			ImageArtifactHash: "",
+			Region:            "test-worker-group",
+		},
+		Task: PartialTask{
+			ProvisionerID: "test-provisioner",
+			Payload: PartialPayload{
+				Command: []string{
+					"taskcluster-proxy",
+					"--version",
+				},
+			},
+		},
+	}
+
+	D2GChainOfTrustHelper(t, &image, []string{}, expected)
+}
+
+func TestD2GChainOfTrustDockerImageArtifact(t *testing.T) {
+
+	setup(t)
+	taskID := CreateArtifactFromFile(t, "docker-images/taskcluster-proxy-v81.0.2.tar.gz", "public/taskcluster-proxy.tar.gz")
+
+	image := d2g.DockerImageArtifact{
+		Path:   "public/taskcluster-proxy.tar.gz",
+		TaskID: taskID,
+		Type:   "task-image",
+	}
+
+	expected := DockerWorkerPartialChainOfTrustCert{
+		Environment: PartialEnvironment{
+			ImageHash:         "sha256:d1c5c701d837b59386563dc702932d200adc41f8bd99a1de451e6fcf378dbfa5",
+			ImageArtifactHash: "sha256:050ba86afcb29779a0df1963df6d1e85eb9feff2c4c2a31b6c559686168be076",
+			Region:            "test-worker-group",
+		},
+		Task: PartialTask{
+			ProvisionerID: "test-provisioner",
+			Payload: PartialPayload{
+				Command: []string{
+					"taskcluster-proxy",
+					"--version",
+				},
+			},
+		},
+	}
+
+	D2GChainOfTrustHelper(t, &image, []string{taskID}, expected)
+}
+
+func TestD2GChainOfTrustIndexedDockerImage(t *testing.T) {
+
+	// TODO: create a task that generates a small docker image, with an index,
+	// and submit it and then modify the task below to pull in that image. The
+	// artifact should expire quickly, because maybe the task is run against a
+	// real taskcluster instance, not necessarily a mock.
```

Comment:
Do you plan on doing this for this PR? Or coming back to it?

Reply:
Probably safe to just remove this test then since, according to @JohanLorenzo and @jcristau chain of trust isn't used with indexed images at all.

Reply:
Coming back to it - among other things it requires adding new mocks in internal/mocktc/index.go so is quite a big job.

Reply:
The issue I see is that it is a feature that is currently not used in production, but could easily be one day (a bit like caches on Windows). I think it is good to have a reminder that we have missing test coverage. It doesn't really make sense to remove the feature, because fuzzing use it (just not in combination with chain of trust).